source/libtomo/misc/CMakeLists.txt

```
@@ -1,41 +1,29 @@
+set(HEADERS "${tomopy_SOURCE_DIR}/include/libtomo/morph.h"
+            "${tomopy_SOURCE_DIR}/include/libtomo/remove_ring.h")
 
-# Locate sources for this project headers are included for IDEs
-file(GLOB misc_tomo_sources
-    ${CMAKE_CURRENT_LIST_DIR}/*.c)
+add_library(tomo-misc SHARED morph.c remove_ring.c ${HEADERS})
 
-# core sources
-set(MISC_LIBRARY_SOURCES ${misc_tomo_sources})
-
-# create library
-add_library(libtomopy-misc SHARED ${MISC_LIBRARY_SOURCES})
+target_include_directories(
+  tomo-misc
+  PUBLIC $<BUILD_INTERFACE:${tomopy_SOURCE_DIR}/include/libtomo>
+         $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include/libtomo>
+  PRIVATE ${CMAKE_CURRENT_LIST_DIR})
 
 # link library to math.h
-if (NOT WIN32)
-    target_link_libraries(libtomopy-misc PUBLIC m)
-endif() 
-
-# where to place the library
-set(OUTPUT_PATH ${CMAKE_BINARY_DIR})
-
-list(APPEND TARGET_INCLUDE_DIRECTORIES "${CMAKE_CURRENT_LIST_DIR}/../../include/misc")
+if(NOT WIN32)
+  target_link_libraries(tomo-misc PUBLIC m)
+endif()
 
-# target properties
-set_target_properties(libtomopy-misc
-    PROPERTIES
-    PREFIX ""
-    RUNTIME_OUTPUT_DIRECTORY    ${OUTPUT_PATH}/tomopy/util/extern
-    LIBRARY_OUTPUT_DIRECTORY    ${OUTPUT_PATH}/tomopy/util/extern
-    ARCHIVE_OUTPUT_DIRECTORY    ${OUTPUT_PATH}/tomopy/util/extern
-    INCLUDE_DIRECTORIES         "${TARGET_INCLUDE_DIRECTORIES}"
-    C_STANDARD                  ${CMAKE_C_STANDARD}
-    C_STANDARD_REQUIRED         ${CMAKE_C_STANDARD_REQUIRED})
+target_compile_definitions(tomo-misc PUBLIC ${${PROJECT_NAME}_DEFINITIONS})
 
-target_compile_definitions(libtomopy-misc PUBLIC
-    ${${PROJECT_NAME}_DEFINITIONS})
+target_compile_options(
+  tomo-misc PUBLIC $<$<COMPILE_LANGUAGE:C>:${${PROJECT_NAME}_C_FLAGS}>)
```

```suggestion
target_compile_options(
  tomo-misc PUBLIC $<BUILD_INTERFACE:$<$<COMPILE_LANGUAGE:C>:${${PROJECT_NAME}_C_FLAGS}>>)
```

`PUBLIC` appears to be used a lot. Since you are not supporting `find_package(libtomo)` you should be very diligent about what is `PUBLIC` because a downstream project "linking" to this target will inherit those flags, definitions, etc. which, especially with compiler flags, can cause build issues if, say, libtomo was built with GCC and the downstream project is using Clang. 

General rule of thumb is make them `PRIVATE` if nothing in the build-tree is linking to this target (or they are specific to that target) or make them `PUBLIC` but wrap it in the `$<BUILD_INTERFACE:...>` if you want another target in the build-tree to inherit those flags/definitions/etc. but they shouldn't be propagated to downstream targets

---

So I should make all of my compile options `PRIVATE` because none of my targets link to each other?

---

Yeah

---

Done.